test/verify/check-upload-component

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/python3 -cimport os, sys; os.execv(os.path.dirname(sys.argv[1]) + "/../common/pywrap", sys.argv)
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+
+import testlib
+from lib.constants import TEST_OS_DEFAULT
+
+
+@testlib.nondestructive
+@testlib.skipBrowser(".upload_files() doesn't work on Firefox", "firefox")
+class TestUploadComponent(testlib.MachineCase):
+    # Test upload component
+    # - Upload progress, slow down network to 3G and test
+    def testBasic(self) -> None:
+        b = self.browser
+        m = self.machine
+
+        self.restore_dir("/home/admin")
+        self.login_and_go("/playground/react-patterns")  # , debugging="upload")
+        files_dir = Path(testlib.TEST_DIR) / "verify" / "files" / "metrics-archives"
+        test_upload_file = str(files_dir / "double_events.zip")
+        dest_file = "/home/admin/double_events.zip"
+
+        filehash = subprocess.check_output(["md5sum", test_upload_file]).strip().decode().split(' ')[0]
```

Comment:
shasum, to not trigger these "meep meep deprecated crypto detected" scanners?

Reply:
Changed to sha512sum.  You can't be too careful these days!